### ISSUE TITLE
dart2js: SyntaxError: Unexpected token &

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#35965]
async closure body methods can have names containing `&`:

```console
$ dart2js main.dart --out=o1.js
$ d8 o1.js
```
 
```console
o1.js:2845: SyntaxError: Unexpected token &
    $call$body$_UU&BB&M_foo_closure: function(a, $async$type) {
                  ^
SyntaxError: Unexpected token &
```

```dart
import 'dart:async';

abstract class II {
  bar();
}

mixin M on II {
  foo<T>() => (T a) async {
    super.bar();
    return [a];
  };
  
}

class BB implements II {
  bar() {
    print('BB.bar');
  }
}

class UU extends BB with M {
}


main() async {
  var uu = UU();
  await uu.foo<int>()(1);
}
```

**Comment by rakudrama** [comment]
Fix in progress: https://dart-review.googlesource.com/c/sdk/+/93465

**Comment by rakudrama** [comment]
Fixed via db71576ef39462a99ab2459c5266f65a431443c5
